src/context/mod.rs

```diff
@@ -339,12 +548,83 @@ pub trait IntoWorkflowClient<'ctx>: Sized {
+/// This pattern is also known as the callback (task token) pattern.
+///
+/// ## Creating awakeables
+///
+/// 1. **Create an awakeable**. This contains a String identifier and a Promise.
+/// 2. **Trigger a task/process** and attach the awakeable ID (e.g., over Kafka, via HTTP, ...).
+///    For example, send an HTTP request to a service that executes the task, and attach the ID to the payload.
+///    You use `ctx.run` to avoid re-triggering the task on retries.
+/// 3. **Wait** until the other process has executed the task.
+///    The handler **receives the payload and resumes**.
+///
+/// ```rust,no_run
+/// # use restate_sdk::prelude::*;
+/// #
+/// # async fn handle(ctx: Context<'_>) -> Result<(), HandlerError> {
+/// /// 1. Create an awakeable
+/// let (id, promise) = ctx.awakeable::<String>();
+///
+/// /// 2. Trigger a task
+/// ctx.run(|| trigger_task_and_deliver_id(id.clone())).await?;
+///
+/// /// 3. Wait for the promise to be resolved
+/// let payload = promise.await?;
+/// # Ok(())
+/// # }
+/// # async fn trigger_task_and_deliver_id(awakeable_id: String) -> Result<(), HandlerError>{
+/// #    Ok(())
+/// # }
+/// ```
+///
 ///
-/// Awakeables can be used to implement external asynchronous systems interactions,
-/// for example you can send a Kafka record including the awakeable id returned by [`ContextAwakeables::awakeable`],
-/// and then let another service consume from Kafka the responses of given external system interaction by using
-/// [`ContextAwakeables::resolve_awakeable`] or [`ContextAwakeables::reject_awakeable`].
+/// ## Completing awakeables
+///
+/// The external process completes the awakeable by either resolving it with an optional payload or by rejecting it
+/// with its ID and a reason for the failure. This throws [a terminal error][crate::errors] in the waiting handler.
+///
+/// - Resolving over HTTP with its ID and an optional payload:
+///
+/// ```text
+/// curl localhost:8080/restate/awakeables/prom_1PePOqp/resolve
+///     -H 'content-type: application/json'
+///     -d '{"hello": "world"}'
+/// ```
+///
+/// - Rejecting over HTTP with its ID and a reason:
+///
+/// ```text
+/// curl localhost:8080/restate/awakeables/prom_1PePOqp/reject
+///     -H 'content-type: text/plain' \
+///     -d 'Very bad error!'
+/// ```
+///
+/// - Resolving via the SDK with its ID and an optional payload, or rejecting with its ID and a reason:
+///
+/// ```rust,no_run
+/// # use restate_sdk::prelude::*;
+/// #
+/// # async fn handle(ctx: Context<'_>, id: String) -> Result<(), HandlerError> {
+/// // Resolve the awakeable
+/// ctx.resolve_awakeable(&id, "hello".to_string());
+///
+/// // Or reject the awakeable
+/// ctx.reject_awakeable(&id, TerminalError::new("my error reason"));
+/// # Ok(())
+/// # }
+/// ```
+///
+/// **Note**: You can return any payload that implements the `serde::Serialize` and `serde::Deserialize` traits ([see serialization docs][crate::serde]).
```

Comment:
This text is honestly super redundant, both here and in the state docs. IMO we don't need to say anything about it here, all the stuff you need to know about ser/de is in the signature of the methods already.

If you really wanna mention it, mention it in the methods docs itself saying "For more info about serialization, see [crate::serde]"